snprc_ehr/src/client/NewAnimalPage/NewAnimalPage.jsx

```diff
@@ -164,25 +192,36 @@ handleSpeciesChange = selectedSpecies => {
         showSpeciesChangeModal: true,
       }))
     } else {
+      let nonPrimate = isNonPrimate(selectedSpecies)
       this.setState(prevState => ({
         ...prevState,
         isDirty: true,
         newAnimalData: {
           ...prevState.newAnimalData,
           species: selectedSpecies,
+          isNonPrimate: nonPrimate,
         },
       }))
       this.loadListsForSpecies(selectedSpecies)
     }
   };
 handleNumAnimalChange = value => {
-    this.setState(
+  const newAssignments = this.state.numAnimals ? this.state.numAnimals : 1
```

Comment:
`const newAssignments = this.state.numAnimals ?? 1`
or
`const newAssignments = this.state.numAnimals || 1`